admin/pg12/do-pg_upgrade.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+set -e
+shopt -s extglob
+
+PGDATA=/var/lib/postgresql/data
+PGDATA_OLD="$PGDATA"/9.5
+PGDATA_NEW="$PGDATA"/12
+PGAMQP_DIR=/tmp/pg_amqp
+
+function cleanup() {
+	if [[ -d "$PGDATA_NEW" ]]; then
+		rm -rf "$PGDATA_NEW"
+	fi
+
+	if [[ -d "$PGDATA_OLD" ]]; then
+		rm -rf "$PGDATA"/!(9.5)
+		mv "$PGDATA_OLD"/* "$PGDATA"
+		rmdir "$PGDATA_OLD"
+	fi
+
+	if [[ -d "$PGAMQP_DIR" ]]; then
+		rm -rf "$PGAMQP_DIR"
+	fi
+}
+trap cleanup EXIT
+
+sudo -u postgres /usr/lib/postgresql/9.5/bin/pg_ctl stop -w -D "$PGDATA" \
+	|| echo 'Assuming server is stopped...'
+
+# We use the --link flag on pg_upgrade below to make hard links instead of
+# copying files, drastically improving the speed of the upgrade. Hard links,
+# of course, require the linked files to be on the same file system, but
+# $PGDATA is the volume *root*. To work around that, we have to move the
+# existing v9.5 cluster to a '9.5' subdir, and create the new v12 cluster in
+# a '12' subdir. Once we're finished, we'll move the new cluster's files
+# back into $PGDATA.
+cd "$PGDATA"
+sudo -u postgres mkdir -p 9.5 12
+chmod 700 9.5 12
+sudo -u postgres mv !(9.5|12) 9.5
+
+sudo -u postgres /usr/lib/postgresql/12/bin/initdb \
+	--encoding utf8 \
+	--username musicbrainz \
+	"$PGDATA_NEW"
+
+git clone https://github.com/omniti-labs/pg_amqp.git "$PGAMQP_DIR"
```

Comment:
We should probably pin a commit here to not pull in changes we don't want.

Reply:
We pinned `1290d7c` in production, which is the latest release `v0.4.1`.

Reply:
Pinned version in https://github.com/metabrainz/musicbrainz-docker/pull/146/commits/01d1db4979b79a4a21ab54ca592f3de860ea37db